lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -1884,6 +1906,46 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     visitPart(RST->getReferentType(), optionalKind);
   }
   
+  Optional<ValueDecl *> findRenamedDecl(const swift::AvailableAttr *AvAttr, const swift::Decl *D) {
+    if (!isa<ValueDecl>(D))
+      return None;
+    
+    SmallVector<ValueDecl *, 4> lookupResults;
+    auto renamedParsedDeclName = swift::parseDeclName(AvAttr->Rename);
+    auto renamedDeclName = renamedParsedDeclName.formDeclName(D->getASTContext());
+    
+    auto declContext = D->getDeclContext();
+    Optional<ValueDecl *>renamedFuncDecl = None;
```

Comment:
Nitpick: This isn't always a function now. :-) You can also probably use `nullptr` as your "not found" value and drop the `Optional`. It's a little more subtle, but a pretty common idiom in our C++ code.